### ISSUE TITLE
Missing package installation (vue-router)

### ISSUE THREAD

**Issue by insukcho** [elastic/apm-agent-rum-js#574]
I'm following [this reference](https://github.com/elastic/apm-agent-rum-js/blob/4.x/docs/vue-integration.asciidoc#installing-elastic-apm-vue-package) to install APM Vue Package.

When I just run `npm install --save @elastic/apm-rum-vue`, I can see below WARN message.

> npm WARN @elastic/apm-rum-vue@0.2.0 requires a peer of vue-router@^3.0.0 but none is installed. You must install peer dependencies yourself.

And it seems [the source code in the same reference](https://www.elastic.co/guide/en/apm/agent/rum-js/master/vue-integration.html#_instrumenting_your_vue_application) is using `vue-router`. Should I run `npm install --save vue-router@^3.0.0` before installing `apm-rum-vue`?

**Comment by vigneshshanmugam** [comment]
The `router` option in the VuePlugin is optional and only needed if you are instrumenting single page applications. 

Though the `warning` message seems to be a bug and we would need to remove `vue-router` from peer dependencies. Thanks for reporting @insukcho

**Comment by insukcho** [comment]
@vigneshshanmugam  Sorry for re-opening the issue. Just want to get clear clarification and might consider doc enhancement.

> The router option in the VuePlugin is optional and only needed if you are instrumenting single page applications.

So, [this reference doc](https://www.elastic.co/guide/en/apm/agent/rum-js/master/vue-integration.html#_instrumenting_your_vue_application) is only for SPA? Entire integration page is only for SPA?

And the Vue integration on non-SPA application does not make sense? Is there any possibility to write Vue code for non-SPA app?

After getting the clarification, I will add this to [our internal self-lab on wiki](https://wiki.elastic.co/display/CC/APM+RUM+JavaScript+Agent+Lab+with+Vue.js) that is for the support engineers who don't have any development background.

**Comment by vigneshshanmugam** [comment]
> And the Vue integration on non-SPA application does not make sense? Is there any possibility to write Vue code for non-SPA app?

Yes absolutely, It would work for non SPA applications as well. the Entire Integrations page is only meant for easy integration of RUM agent with respective frameworks. Plus if there is any routing available for the application, It would be easier for us to instrument the soft navigation (routing) and use the respective path name of the route as the transaction name. 

Does this make sense?

**Comment by insukcho** [comment]
Yeh, make sense to me. I'm just trying to understand `vue-router` exactly.

Honestly, I'm not familiar with this but it seems using `vue-router` is just an option.

So current reference code example is like below.
```
import Vue from 'vue'
import VueRouter from 'vue-router'
import { ApmVuePlugin } from '@elastic/apm-rum-vue'

const Home = { template: '<div>home</div>' }

Vue.use(VueRouter)

const router = VueRouter({
  mode: 'history',
  routes: [
    { path: '/', component: Home }
  ]
})

Vue.use(ApmVuePlugin, {
  router,
  config: {
    serviceName: 'app-name',
    // agent configuration
  }
})

// app specific code
```

If I remove all `vue-router` related code, it will look like below.
```
import Vue from 'vue'
import { ApmVuePlugin } from '@elastic/apm-rum-vue'

Vue.use(ApmVuePlugin, {
  config: {
    serviceName: 'app-name',
    // agent configuration
  }
})

// app specific code
```

Do you think adding this example in reference doc will be worthwhile? Or any Vue developer will be not confused even they are not using `vue-router`? (maybe, not using 'vue-router' doesn't make sense to them, though)

**Comment by vigneshshanmugam** [comment]
Actually the idea of making the vue-router optional was from this PR which is contributed by a community member https://github.com/elastic/apm-agent-rum-js/pull/553 and it did completely make sense since the user would be willing to use the ApmVuePlugin instead of using the generic `@elastic/apm-rum` which provides better integration with vue apps. We can also improve the document by adding an example without vue-router. 

I am open to ideas here. Let me know what you think would avoid the confusion.

**Comment by insukcho** [comment]
Sorry for late response @vigneshshanmugam 

> We can also improve the document by adding an example without vue-router.

Sounds great to me. Let's do it!
